components/Typography/src/MDCTypography.m

```diff
@@ -185,45 +186,135 @@ + (UIFont *)boldFontFromFont:(UIFont *)font {
 
 @end
 
+@interface MDCSystemFontLoader ()
+
+/**
+ Basic cache to keep track of recently-used fonts.
+ */
+@property(nonatomic, strong) NSCache *fontCache;
+
+@end
+
 @implementation MDCSystemFontLoader
 
+- (instancetype)init {
+  self = [super init];
+  if (self) {
+    _fontCache = [[NSCache alloc] init];
+    [[NSNotificationCenter defaultCenter] addObserver:self
+                                             selector:@selector(didChangeContentSizeCategory)
+                                                 name:UIContentSizeCategoryDidChangeNotification
+                                               object:nil];
```

Comment:
In the past, this was the only notification I could find related to dynamic type.  We don't actually need it (our fonts don't change with dynamic type changes), so it's an unnecessary "safety" measure.  I could consider removing the notification observer as well, since there isn't a clean lifecycle for what is "usually" a singleton (the font loader).